presto-clp/src/test/java/com/facebook/presto/plugin/clp/TestClpFilterToKql.java

```diff
@@ -14,194 +14,206 @@
 
-        testFilter("city.Name IN ('hello world', 'hello world 2')", "(city.Name: \"hello world\" OR city.Name: \"hello world 2\")", null, sessionHolder);
+        testFilter(tryPushDown("city.Name IN ('hello world', 'hello world 2')", sessionHolder), "(city.Name: \"hello world\" OR city.Name: \"hello world 2\")", null, sessionHolder);
     }
 
     @Test
-    public void testIsNullPushdown()
+    public void testIsNullPushDown()
     {
         SessionHolder sessionHolder = new SessionHolder();
 
-        testFilter("city.Name IS NULL", "NOT city.Name: *", null, sessionHolder);
-        testFilter("city.Name IS NOT NULL", "NOT NOT city.Name: *", null, sessionHolder);
-        testFilter("NOT (city.Name IS NULL)", "NOT NOT city.Name: *", null, sessionHolder);
+        testFilter(tryPushDown("city.Name IS NULL", sessionHolder), "NOT city.Name: *", null, sessionHolder);
+        testFilter(tryPushDown("city.Name IS NOT NULL", sessionHolder), "NOT NOT city.Name: *", null, sessionHolder);
+        testFilter(tryPushDown("NOT (city.Name IS NULL)", sessionHolder), "NOT NOT city.Name: *", null, sessionHolder);
     }
 
     @Test
-    public void testComplexPushdown()
+    public void testComplexPushDown()
     {
         SessionHolder sessionHolder = new SessionHolder();
 
-        testFilter(
-                "(fare > 0 OR city.Name like 'b%') AND (lower(city.Region.Name) = 'hello world' OR city.Name IS NULL)",
-                "((fare > 0 OR city.Name: \"b*\"))",
-                "(lower(city.Region.Name) = 'hello world' OR city.Name IS NULL)",
-                sessionHolder);
-        testFilter(
-                "city.Region.Id = 1 AND (fare > 0 OR city.Name NOT like 'b%') AND (lower(city.Region.Name) = 'hello world' OR city.Name IS NULL)",
-                "((city.Region.Id: 1 AND (fare > 0 OR NOT city.Name: \"b*\")))",
-                "lower(city.Region.Name) = 'hello world' OR city.Name IS NULL",
-                sessionHolder);
+        testFilter(tryPushDown("(fare > 0 OR city.Name like 'b%') AND (lower(city.Region.Name) = 'hello world' OR city.Name IS NULL)", sessionHolder), "((fare > 0 OR city.Name: \"b*\"))", "(lower(city.Region.Name) = 'hello world' OR city.Name IS NULL)", sessionHolder);
+        testFilter(tryPushDown("city.Region.Id = 1 AND (fare > 0 OR city.Name NOT like 'b%') AND (lower(city.Region.Name) = 'hello world' OR city.Name IS NULL)", sessionHolder), "((city.Region.Id: 1 AND (fare > 0 OR NOT city.Name: \"b*\")))", "lower(city.Region.Name) = 'hello world' OR city.Name IS NULL", sessionHolder);
     }
 
-    private void testFilter(String sqlExpression, String expectedKqlExpression, String expectedRemainingExpression, SessionHolder sessionHolder)
+    public void testMetadataSqlGeneration()
+    {
+        SessionHolder sessionHolder = new SessionHolder();
+        testFilterWithMetadataSql(
+                tryPushDown("(fare > 0 AND city.Name like 'b%')",
+                        sessionHolder),
```

Comment:
```suggestion
                tryPushDown("(fare > 0 AND city.Name like 'b%')", sessionHolder),
```
Same for below usages